doc/contribute/devtheory.rst

```diff
@@ -0,0 +1,185 @@
+Developer's Theory Guide
+========================
+
+This guide is meant to give developers a comprehensive understanding of how
+DAGMC works as it exists in MOAB and the key components one may use during
+development.
+
+Contructor
+~~~~~~~~~~~~
+
+To initialize a DAGMC instance, the constructor must be called. If the
+constructor is called with no arguments, then a new instance is created. Else,
```

Comment:
Okay, I think I made the proper clarification, but I am unsure about what exactly is happening. Let me know if how I wrote it up now is correct.